processing/src/main/java/org/apache/druid/query/scan/ScanQueryLimitRowIterator.java

```diff
@@ -70,9 +93,14 @@ public boolean hasNext()
   @Override
   public ScanResultValue next()
   {
-    ScanResultValue batch = yielder.get();
-    if (ScanQuery.RESULT_FORMAT_COMPACTED_LIST.equals(resultFormat) ||
-        ScanQuery.RESULT_FORMAT_LIST.equals(resultFormat)) {
+    if (ScanQuery.ResultFormat.RESULT_FORMAT_VALUE_VECTOR.equals(resultFormat)) {
+      throw new UOE(ScanQuery.ResultFormat.RESULT_FORMAT_VALUE_VECTOR + " is not supported yet");
+    }
+
+    // We want to perform batching if we are not time-ordering or are at the outer level if we are re time-ordering
+    if (query.getOrder() == ScanQuery.Order.NONE ||
+        !query.getContextBoolean(ScanQuery.CTX_KEY_OUTERMOST, true)) {
```

Comment:
The  `!query.getContextBoolean(ScanQuery.CTX_KEY_OUTERMOST, true)` needs to be flipped